plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -1018,6 +1022,104 @@ a_enum_2 Enum('hello', 'world', 'a', 'b', 'c'))
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE 'b'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE 'b%'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE 'b%'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b%'")).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '%b%'" + withConnectorExpression)).isFullyPushedDown();
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 NOT LIKE '%b%'")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_enum_1 NOT LIKE '%b%'" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE unsupported_1 NOT LIKE '%b%'")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE unsupported_1 NOT LIKE '%b%'" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE a_string_alias")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE a_string_alias" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE a_enum_1")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE a_enum_1" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE unsupported_1")).isNotFullyPushedDown(FilterNode.class);
+            assertThat(query(convertToVarchar, "SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE unsupported_1" + withConnectorExpression)).isNotFullyPushedDown(FilterNode.class);
+            // metacharacters
+            assertThat(query("SELECT some_column FROM " + table.getName() + " WHERE a_string NOT LIKE '_'")).isFullyPushedDown();
```

Comment:
Can we have a dedicated test method for `LIKE` I think this test method is being overloaded a lot

Reply:
We can. However I prefer to generate test cases `operation * type * data`, leaving only some basic smoke checks explicit.
I have a plan to do it after implementing varchars pushdown, as it maybe considered as controversial.

Reply:
Since `LIKE` is not a part of tuple domain - We can have tests with additional `OR` for a single case and not with all conditions

Reply:
as now patterns which contain `\` are excluded from pushdown, having explicit tests even when condition is processed by TupleDomain maybe more desired.